src/duk_debugger.c

```diff
@@ -37,8 +37,23 @@ DUK_LOCAL void duk__debug_do_detach1(duk_heap *heap) {
 	 * inside the detached callback.
 	 */
 
+	if (heap->dbg_detaching) {
+		return;
+	}
+
 	DUK_D(DUK_DPRINT("debugger transport detaching, marking transport broken"));
 
+	heap->dbg_detaching = 1;
+
+	/* avoid sending multiple Detach notifys */
+	if (heap->dbg_write_cb != NULL) {
+		duk_hthread *thr = heap->heap_thread;
+
```

Comment:
Oh, okay.  I assumed `heap_thread` would always be non-NULL by definition so didn't add the assert.